src/lib.rs

```diff
@@ -346,10 +345,425 @@ impl From<Error> for io::Error {
+                self.clone().into(),
+            ))
+    }
+
+    /// Returns the path without its final component, if there is one.
+    ///
+    /// The same as [`std::path::Path::parent()`], except that it returns a
+    /// `Result` with a rich [`path_abs::Error`] when a problem is encountered.
+    ///
+    /// [`path_abs::Error`]: struct.Error.html
+    /// [`std::path::Path::parent()`]: https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.parent
+    fn parent(&self) -> Result<&Path> {
+        let parent_path = Path::parent(self.borrow());
+        if let Some(p) = parent_path {
+            Ok(p)
+        } else {
+            Err(Error::new(
+                io::Error::new(io::ErrorKind::NotFound, "path has no parent"),
+                "truncating to parent",
+                self.clone().into(),
+            ))
+        }
+    }
+}
+
+impl<T> PathInfo for T where T: Clone + Borrow<PathBuf> + Into<Arc<PathBuf>> {}
+
+/// Methods that modify a path.
+///
+/// These methods are not implemented for all `path_abs` types because they
+/// may break the type's invariant. For example, if you could call
+/// `truncate_to_parent()` on a `PathFile`, it would no longer be the path to
+/// a file, but the path to a directory.
+///
+/// As a general rule, methods that can return an error will return a rich
+/// [`path_abs::Error`] instead of a [`std::io::Error`].
+pub trait PathMut: PathInfo {
+    /// Appends `path` to this path.
+    ///
+    /// Note that this method represents pure concatenation, not "adjoinment"
+    /// like [`PathBuf::push`].
+    ///
+    /// # Errors
+    ///
+    /// This method returns an error if the `path` parameter contains a prefix
+    /// component (on Windows, like `C:`), or enough `..` components to consume
+    /// this path.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use std::path::Path;
+    /// use path_abs::PathMut;
+    ///
+    /// let mut somepath = Path::new("foo").to_path_buf();
+    /// somepath.append("bar");
+    ///
+    /// assert_eq!(somepath.as_path(), Path::new("foo/bar"));
+    /// ```
+    ///
+    /// [`PathBuf::push`]: https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#method.push
+    fn append<P: AsRef<Path>>(&mut self, path: P) -> Result<()>;
+
+    /// Removes the last component of this path.
+    ///
+    /// # Errors
+    ///
+    /// This method returns an error if this path has no parent, i.e. it
+    /// represents a filesystem root, like `/` on Unix or `C:\` on Windows.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # fn example() -> Result<(), path_abs::Error> {
+    /// use std::path::Path;
+    /// use path_abs::PathMut;
+    ///
+    /// let executable = Path::new("/usr/loca/bin/myapp");
+    /// let mut install_path = executable.to_path_buf();
+    /// install_path.truncate_to_parent()?;
+    ///
+    /// assert_eq!(install_path.as_path(), Path::new("/usr/local/bin"));
+    /// # Ok(()) }
```

Comment:
several of these blocks don't have closing \`\`\`

Reply:
Should be fixed in bdc4dad